Bottom: d22e6e345c1423daf9c06f59c0b09fd16c64e371
Top:    5625997313f89f745a58d9549c1414195e478c6a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-05-10 17:28:47 +0100

Fix crash hovering over unconnected element


---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 7466e59..1930bd9 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -754,17 +754,23 @@ describe_location (LocationType X, LocationType Y)
   void *ptr1, *ptr2, *ptr3;
   int type;
   int Range = 0;
-  char *elename;
+  char *elename = "";
   char *pinname;
   char *netname = NULL;
   char *description;
 
   /* check if there are any pins or pads at that position */
 
-  type = SearchObjectByLocation (LOOKUP_FIRST,
+  type = SearchObjectByLocation (PIN_TYPE | PAD_TYPE,
                                  &ptr1, &ptr2, &ptr3, X, Y, Range);
   if (type == NO_TYPE)
-    type = SearchObjectByLocation (LOOKUP_MORE,
+    type = SearchObjectByLocation (VIA_TYPE | LINE_TYPE | ARC_TYPE,
+                                   &ptr1, &ptr2, &ptr3, X, Y, Range);
+  if (type == NO_TYPE)
+    type = SearchObjectByLocation (RATLINE_TYPE,
+                                   &ptr1, &ptr2, &ptr3, X, Y, Range);
+  if (type == NO_TYPE)
+    type = SearchObjectByLocation (POLYGON_TYPE,
                                    &ptr1, &ptr2, &ptr3, X, Y, Range);
 
   if (type == NO_TYPE)
@@ -775,7 +781,9 @@ describe_location (LocationType X, LocationType Y)
       GetLayerNumber (PCB->Data, (LayerTypePtr) ptr1) >= max_layer)
     return NULL;
 
-  elename = UNKNOWN (NAMEONPCB_NAME ((ElementTypePtr) ptr1));
+  if (type == PIN_TYPE || type == PAD_TYPE)
+    elename = UNKNOWN (NAMEONPCB_NAME ((ElementTypePtr) ptr1));
+
   pinname = ConnectionName (type, ptr1, ptr2);
 
   if (pinname == NULL)
